cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -128,15 +128,16 @@ func checkForNodeNameDuplicates(clientSet *clientset.Clientset) error {
 	if err != nil {
 		return fmt.Errorf("Failed to get node hostname [%v]", err)
 	}
-	nodeList, err := clientSet.Nodes().List(v1.ListOptions{})
+	node, err := clientSet.Nodes().Get(hostName, v1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("Failed to list the nodes in the cluster: [%v]\n", err)
-	}
-	for _, node := range nodeList.Items {
-		if hostName == node.Name {
-			return fmt.Errorf("Node with name [%q] already exists.", node.Name)
+		errstring := fmt.Sprintf("nodes \"%s\" not found", hostName)
```

Comment:
Instead of comparing the error string, use the kubernetes error package IsNotFound utility method that does type checking